static/app/components/events/interfaces/spans/spanProfileDetails.tsx

```diff
@@ -0,0 +1,194 @@
+
+import {SpanType} from './types';
+
+interface SpanProfileDetailsProps {
+  event: Readonly<EventTransaction>;
+  span: Readonly<SpanType>;
+}
+
+export function SpanProfileDetails({event, span}: SpanProfileDetailsProps) {
+  const profileGroup = useProfileGroup();
+
+  // TODO: Pick another thread if it's more relevant.
+  const threadId = useMemo(
+    () => profileGroup.profiles[profileGroup.activeProfileIndex]?.threadId,
+    [profileGroup]
+  );
+
+  const profile = useMemo(() => {
+    if (!defined(threadId)) {
+      return null;
+    }
+    return profileGroup.profiles.find(p => p.threadId === threadId) ?? null;
+  }, [profileGroup.profiles, threadId]);
+
+  const nodes: CallTreeNode[] = useMemo(() => {
+    if (profile === null) {
+      return [];
+    }
+
+    const relativeStartTimestamp = formatTo(
+      span.start_timestamp - event.startTimestamp,
+      'second',
+      profile.unit
+    );
+    const relativeStopTimestamp = formatTo(
+      span.timestamp - event.startTimestamp,
+      'second',
+      profile.unit
+    );
+
+    return getTopNodes(profile, relativeStartTimestamp, relativeStopTimestamp);
+  }, [profile, span, event]);
+
+  const frames = useMemo(() => {
+    if (!nodes.length) {
+      return [];
+    }
+
+    return extractFrames(nodes[0], event.platform || 'other');
+  }, [event, nodes]);
+
+  if (!defined(profile)) {
+    return null;
+  }
+
+  if (!nodes.length) {
+    return null;
+  }
+
+  return (
+    <StackTrace
+      event={event}
+      hasHierarchicalGrouping
+      newestFirst
+      platform={event.platform || 'other'}
+      stacktrace={{
+        framesOmitted: null,
+        hasSystemFrames: false,
+        registers: null,
+        frames,
+      }}
+      nativeV2
+      stackView={STACK_VIEW.APP}
+    />
+  );
+}
+
+function getTopNodes(profile: Profile, startTimestamp, stopTimestamp): CallTreeNode[] {
+  let duration = 0;
+
+  const callTree: CallTreeNode = new CallTreeNode(ProfilingFrame.Root, null);
+
+  for (const sample of profile.samples) {
+    // TODO: should this take self times into consideration?
+    const inRange = startTimestamp <= duration && duration < stopTimestamp;
+
+    duration += sample.selfWeight;
+
+    if (sample.isRoot() || !inRange) {
+      continue;
+    }
+
+    const stack: CallTreeNode[] = [sample];
+    let node: CallTreeNode | null = sample;
+
+    while (node && node.parent && !node.parent.isRoot()) {
+      node = node.parent;
+      stack.push(node);
+    }
+
+    let tree = callTree;
+
+    // make sure to iterate the stack backwards here, the 0th index is the
+    // inner most frame, and the last index is the outer most frame
+    for (let i = stack.length - 1; i >= 0; i--) {
+      node = stack[i]!;
+      const frame = node.frame;
+
+      // find a child in the current tree with the same frame,
+      // merge the current node into it if it exists
+      let last = tree.children.find(n => n.frame === frame);
```

Comment:
Since samples are ordered, I think you can just check if the frames of the last tree child match instead of the entire collection

Reply:
That is not true here since we're merging stacks within the span regardless of order.